filters/auth/oidc_test.go

```diff
@@ -549,6 +554,87 @@ func TestCreateFilterOIDC(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestCreateFilterOIDC_subdomainsToRemove(t *testing.T) {
```

Comment:
TestOIDCSetup has already huge complexity. Adding domain setting verification into this test case will increase complexity even more. We only have to test whether parameter is set properly and you use same approach with the Test_tokenOidcFilter_getMaxAge already. That's the reasons why I introduced a new test case. However if you still prefer to bloat TestOIDCSetup  I will do this.

Reply:
> you use same approach with the Test_tokenOidcFilter_getMaxAge already
Cookie age is verified in `TestOIDCSetup` though. 

We need to verify that cookie has an expected domain and we do not need to cover all the cases maybe as `TestExtractDomainFromHost` does that already.

Reply:
Just tried to implement a check in TestOIDCSetup, however I am afraid it is not possible to provide a custom hostname for the callback call instead of ip address. Any suggestions?

Reply:
Maybe you can try to use `localhost` as TLD.
On my machine this works and could be used for testing (replace host part of the listener URL if there is a configured domain for the purpose:
```
% host foo.bar.localhost                                                                                                                                                                dev
foo.bar.localhost has address 127.0.0.1
foo.bar.localhost has IPv6 address ::1
```

Reply:
> Just tried to implement a check in TestOIDCSetup, however I am afraid it is not possible to provide a custom hostname for the callback call instead of ip address. Any suggestions?

It is possible but the problem is that it should resolve to the test server address (e.g. 127.0.0.1) and the only one that is universal is `localhost` but it does not support subdomains afaik.

> Maybe you can try to use localhost as TLD.
> On my machine this works

Works on Ubuntu as well but afaik `localhost` can not have subdomains https://stackoverflow.com/questions/19016553/add-subdomain-to-localhost-url so maybe it is due to `systemd-resolved` https://unix.stackexchange.com/questions/401966/how-does-every-subdomain-of-localhost-point-to-localhost-on-fedora

Reply:
It is possible like so
```diff
diff --git a/filters/auth/oidc_test.go b/filters/auth/oidc_test.go
index a7e505b6..27a5c613 100644
--- a/filters/auth/oidc_test.go
+++ b/filters/auth/oidc_test.go
@@ -731,7 +731,11 @@ func TestOIDCSetup(t *testing.T) {
                                t.Errorf("Failed to parse url %s: %v", proxy.URL, err)
                        }
 
-                       oidcServer := createOIDCServer(proxy.URL+"/redirect", tc.client, tc.clientsecret)
+                       redirectURL, _ := url.Parse(proxy.URL)
+                       redirectURL.Host = "localhost:" + redirectURL.Port()
+                       redirectURL.Path = "/redirect"
+
+                       oidcServer := createOIDCServer(redirectURL.String(), tc.client, tc.clientsecret)
                        defer oidcServer.Close()
                        t.Logf("oidc/auth server URL: %s", oidcServer.URL)
 
@@ -739,7 +743,7 @@ func TestOIDCSetup(t *testing.T) {
                        sargs := []interface{}{
                                tc.client,
                                tc.clientsecret,
-                               proxy.URL + "/redirect",
+                               redirectURL.String(),
                        }
 
                        // test that, we get an error if provider is no url
@@ -827,7 +831,7 @@ func TestOIDCSetup(t *testing.T) {
                        t.Logf("Got body: %s", bs)
 
                        // make sure cookies have correct maxAge according to exp
-                       for _, c := range client.Jar.Cookies(reqURL) {
+                       for _, c := range client.Jar.Cookies(redirectURL) {
                                maxAge := time.Duration(c.MaxAge) * time.Second
                                assert.True(t, maxAge >= 2*time.Hour-time.Minute && maxAge <= 2*time.Hour,
                                        "maxAge has to be within [2h - 1m, 2h]")
```

It also seems that cookie testing is not really strict, e.g. there is a test case where cookies are not expected and test does not check that, also it uses `reqURL` but it only works because redirect url host is IP address (or localhost in above) and hence cookie domain is IP (or localhost) as well.

The proper cookie testing should expect (or expect no) cookies and check domain which in case of multi-level hostname would not be equal to the `reqURL` or `redirectURL` due to subdomain removal.

Reply:
I guess we could hack http clients (also OAuth2 client) to resolve any desired domain to 127.0.0.1 but that needs thinking and refactoring.
@szuecs  Should we leave cookie domain testing out of scope here until we find a way to do it properly, wdyt?

Reply:
I am fine with both. So either check it in the client, but this needs a custom Dial() or DialContext() in the http.Transport as far as I remember or omit this part in testing and create an issue to postpone it.

Reply:
I guess you can always make it if the client supports reading `/etc/hosts`, but systemd-resolved seems to be ok for automated tests. I would just try it and see what GH actions (and for maintainers CDP) will tell us about.

Reply:
I have opened an issue https://github.com/zalando/skipper/issues/1926 and will have a look into it.

Reply:
thanks!

Reply:
The https://github.com/zalando/skipper/pull/1927 was merged and now it should be possible to test cookie domain. Please rebase to see example tests.

Reply:
This test creates createOIDCServer that is not used.

Lets remove this test and add a test or two to the `TestOIDCSetup` integration test because we are still not testing domain handling when new option is different from the default.